File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/tenant/TenantRepository.java
package br.com.deltaglobalbank.identity.domain.tenant;

import java.util.UUID;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TenantRepository {
    Tenant findById(UUID id);

    Tenant findBySlug(String slug);

    boolean existsBySlug(String slug);

    Page<Tenant> findPage(Pageable pageable);

    Tenant save(Tenant tenant);

    void delete(UUID id);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/tenant/TenantRepository.kt
package br.com.deltaglobalbank.identity.domain.tenant

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.UUID

interface TenantRepository {
    fun findById(id: UUID): Tenant?
    fun findBySlug(slug: String): Tenant?
    fun existsBySlug(slug: String): Boolean
    fun findPage(pageable: Pageable): Page<Tenant>
    fun save(tenant: Tenant): Tenant
    fun delete(id: UUID)
}
